projects/vdk-core/src/vdk/internal/core/errors.py

```diff
@@ -399,6 +410,8 @@ def find_whom_to_blame_from_exception(exception: Exception) -> ResolvableBy:
         )  # TODO find out if this is a local or platform deployment and fix this line.
     if issubclass(type(exception), PlatformServiceError):
         return ResolvableBy.PLATFORM_ERROR
+    if issubclass(type(exception), CancelJobExecutionException):
+        return ResolvableBy.USER_ERROR
```

Comment:
I don't like this. CancelJobExecution (or cancel remaining steps) is supposed to be "success" as written in the doc. 

We are using exception for control flow which is anti pattern as very well argumented [here](https://dzone.com/articles/exceptions-as-controlflow-in-java) or [here](https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why) or [here](https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl).  And the bad side effects show up here.

But in our case I don't think we have another "lean" solution. But we can reduce the scope. 

I think this check might not be necessary if we handle it in "run_step" and return simply status (as I wrote in above comment).  Then nobody else but run_step would know the exception is raised ...